src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallerFactory.cs

```diff
@@ -20,7 +22,32 @@ internal class WorkloadInstallerFactory
             string dotnetDir = null, 
             PackageSourceLocation packageSourceLocation = null)
         {
+            var installType = GetWorkloadInstallType(sdkFeatureBand);
+
+            if (installType == InstallType.Msi && OperatingSystem.IsWindows())
+            {
+                return new NetSdkMsiInstaller();
+            }
+
             return new NetSdkManagedInstaller(reporter, sdkFeatureBand, workloadResolver, nugetPackageDownloader, verbosity: verbosity, dotnetDir: dotnetDir, packageSourceLocation: packageSourceLocation);
         }
+
+        /// <summary>
+        /// Determines the <see cref="InstallType"/> associated with a specific SDK version.
+        /// </summary>
+        /// <param name="sdkFeatureBand">The SDK version to check.</param>
+        /// <returns>The <see cref="InstallType"/> associated with the SDK.</returns>
+        public static InstallType GetWorkloadInstallType(SdkFeatureBand sdkFeatureBand)
+        {
+            string installerTypePath = Path.Combine(Path.GetDirectoryName(Environment.ProcessPath), "metadata",
```

Comment:
Can we use the dotnetDir parameter [here](https://github.com/dotnet/sdk/pull/17926/files#diff-30590c656d6aa0fc5b561cb5ef14bd838f7ade7aa2e7c008cda0aa2cf4f79c5aR22) instead of checking the process path? This will make testing easier

Reply:
Yes, we should